server/src/main/java/org/elasticsearch/common/blobstore/BlobStore.java

```diff
@@ -29,6 +32,14 @@ public interface BlobStore extends Closeable {
      */
     void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames) throws IOException;
 
+    default void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames, @Nullable Purpose purpose) throws IOException {
```

Comment:
I would also much prefer not to have an implicit default for the `purpose` parameter like this. Experience shows that we will forget to set it in future work unless the caller is required to make an explicit choice.

That said, temporarily adding an implicit default is a reasonable strategy for avoiding breaking the serverless build, but it must only be temporary. Ideally we'd mark the overloads that don't take the `Purpose` as `@Deprecated(forRemoval = true)`.

Reply:
Agreed especially since this interface method is pretty new. I will go through the deprecation and removal process once we are happy with the proposed approach in this PR.